# Observing the Immobiliare market in Belgium

### Subtitle

Image

## Introduction

Describe the project and define the questions

## The process

1. Scraped **Immoweb** information of houses and apartments
2. Explored the data to understand the relationship between the variables and the price of the different types of elements

Used Jupyter Lab, description of all the modules

For the purpose of this exercise, an **element** can be a
house, apartment or a building

### 1. Scraping

##### Get links for each element

##### Get raw attributes for each element

##### Saving the information into a CSV

### 2. Cleaning Data

In [39]:
import pandas as pd

##### Transform the CSV into a DataFrame

In [51]:
df = pd.read_csv('data/all_data_frame.csv',low_memory=False)

DtypeWarning: Columns (24,25,26,27,72,73,74,75,106,115,123) have mixed types

##### Cleaning the DataFrame

In [15]:
df.drop_duplicates()

Available as of                Available date  \
0       After signing the deed   November 12 2022 - 12:00 AM   
1       After signing the deed  September 15 2022 - 12:00 AM   
2       After signing the deed                           NaN   
3       After signing the deed                           NaN   
4      Depending on the tenant                           NaN   
...                        ...                           ...   
18474                      NaN                           NaN   
18475                      NaN                           NaN   
18476                      NaN                           NaN   
18477                      NaN                           NaN   
18478                      NaN                           NaN   

       Construction year Building condition Street frontage width  \
0                 1899.0               Good                   5 m   
1                 1941.0               Good                4.44 m   
2                 1961.0        To renovate                   7 m   
3                 1989.0             As new                  12 m   
4                 1924.0      To be done up                   7 m   
...                  ...                ...                   ...   
18474                NaN             As new                   NaN   
18475                NaN                NaN                   NaN   
18476                NaN                NaN                   NaN   
18477                NaN                NaN                   NaN   
18478             2022.0                NaN                   NaN   

       Number of frontages                          Surroundings type  \
0                      2.0                                   Isolated   
1                      2.0                                      Urban   
2                      3.0  Living area (residential, urban or rural)   
3                      4.0                                  Landscape   
4                      2.0                                        NaN   
...                    ...                                        ...   
18474                  NaN                                        NaN   
18475                  2.0  Living area (residential, urban or rural)   
18476                  2.0  Living area (residential, urban or rural)   
18477                  2.0  Living area (residential, urban or rural)   
18478                  NaN                                        NaN   

                   Virtual visit           Living area  Living room surface  \
0      Link to the virtual visit  122 m² square meters  14 m² square meters   
1                            NaN  119 m² square meters  32 m² square meters   
2                            NaN  200 m² square meters  45 m² square meters   
3                            NaN  404 m² square meters  55 m² square meters   
4      Link to the virtual visit  350 m² square meters                  NaN   
...                          ...                   ...                  ...   
18474                        NaN  126 m² square meters                  NaN   
18475                        NaN   84 m² square meters                  NaN   
18476                        NaN   80 m² square meters                  NaN   
18477                        NaN   90 m² square meters                  NaN   
18478                        NaN  146 m² square meters                  NaN   

       ... Current monthly revenue Percentage rented Venue of the sale  \
0      ...                     NaN               NaN               NaN   
1      ...                     NaN               NaN               NaN   
2      ...                     NaN               NaN               NaN   
3      ...                     NaN               NaN               NaN   
4      ...                     NaN               NaN               NaN   
...    ...                     ...               ...               ...   
18474  ...                     NaN               NaN               NaN   

**List of raw attributes**

- **Col1 Available as of** : DataType: string, Values: ['After signing the deed', 'Depending on the tenant', 'To be defined', 'Immediately', Nan ,'At delivery']  Comments: **not usefull**
- **Col2 Available date** : DataType: date Month day Year, Comments: not usefull
- **Col3 Construction year** : DataType: integer ####
- **Col4 Building condition** : DataType: string, Values: ['Good', 'To renovate', 'As new', 'To be done up', nan, 'Just renovated', 'To restore'] Comments: **usefull**
- **Col5 Street frontage width** : DataType: string, Values: #.## m, Comments: need to remove ' m' new DataType: float64, Comments: **not usefull**
- **Col6 Number of frontages** :
- **Col7 Surroundings type** :
- **Col8 Virtual visit** :
- **Col9 Living area** :
- **Col10 Living room surface** :
- **Col11 Dining room** :
- **Col12 Kitchen type** :
- **Col13 Kitchen surface** :
- **Col14 Bedrooms** :
- **Col15 Bedroom 1 surface** :
- **Col16 Bedroom 2 surface** :
- **Col17 Bathrooms** :
- **Col18 Shower rooms** :
- **Col19 Toilets** :
- **Col20 Office surface** :
- **Col21 Office** :
- **Col22 Basement** :
- **Col23 Attic** :
- **Col24 Furnished** :
- **Col25 Surface of the plot** :
- **Col26 Width of the lot on the street** :
- **Col27 Connection to sewer network** :
- **Col28 Gas, water &amp; electricity** :
- **Col29 Garden surface** :
- **Col30 Terrace surface** :
- **Col31 Elevator** :
- **Col32 Accessible for disabled people** :
- **Col33 TV cable** :
- **Col34 Swimming pool** :
- **Col35 Primary energy consumption** :
- **Col36 Energy class** :
- **Col37 Unnamed: 36** :
- **Col38 Reference number of the EPC report** :
- **Col39 CO₂ emission** :
- **Col40 Yearly theoretical total energy consumption** :
- **Col41 Heating type** :
- **Col42 Double glazing** :
- **Col43 Planning permission obtained** :
- **Col44 Total ground floor buildable** :
- **Col45 Subdivision permit** :
- **Col46 Possible priority purchase right** :
- **Col47 Proceedings for breach of planning regulations** :
- **Col48 Flood zone type** :
- **Col49 Latest land use designation** :
- **Col50 Price** :
- **Col51 Cadastral income** :
- **Col52 Tenement building** :
- **Col53 Address** :
- **Col54 Website** :
- **Col55 External reference** :
- **Col56 subtype_of_property** :
- **Col57 type_of_property** :
- **Col58 type_of_sale** :
- **Col59 locality** :
- **Col60 zip_code** :
- **Col61 url** :
- **Col62 Bedroom 3 surface** :
- **Col63 Bedroom 4 surface** :
- **Col64 Neighbourhood or locality** :
- **Col65 Covered parking spaces** :
- **Col66 Outdoor parking spaces** :
- **Col67 Dressing room** :
- **Col68 Laundry room** :
- **Col69 Professional space** :
- **Col70 Basement surface** :
- **Col71 Attic surface** :
- **Col72 Armored door** :
- **Col73 Land is facing street** :
- **Col74 Wooded land** :
- **Col75 Plot at rear** :
- **Col76 Flat land** :
- **Col77 Caretaker** :
- **Col78 Intercom** :
- **Col79 Secure access / alarm** :
- **Col80 Air conditioning** :
- **Col81 Visio phone** :
- **Col82 Jacuzzi** :
- **Col83 Sauna** :
- **Col84 Internet** :
- **Col85 Conformity certification for fuel tanks** :
- **Col86 Heat pump** :
- **Col87 Photovoltaic solar panels** :
- **Col88 Thermic solar panels** :
- **Col89 Common water heater** :
- **Col90 Type of building** :
- **Col91 Terrace** :
- **Col92 Bedroom 5 surface** :
- **Col93 Professional space surface** :
- **Col94 As built plan** :
- **Col95 Living room** :
- **Col96 How many fireplaces?** :
- **Col97 Garden** :
- **Col98 Garden orientation** :
- **Col99 Terrace orientation** :
- **Col100 E-level (overall energy performance)** :
- **Col101 Agent's name** :
- **Col102 Number of annexes** :
- **Col103 Isolated** :
- **Col104 Obligation to build** :
- **Col105 Monthly charges** :
- **Col106 Property name** :
- **Col107 Extra information** :
- **Col108 Lump sum** :
- **Col109 Monthly annuity** :
- **Col110 Indexed annuity** :
- **Col111 Reversionary annuity** :
- **Col112 Bare ownership sale** :
- **Col113 Number of annuitants** :
- **Col114 Age of annuitant** :
- **Col115 Maximum duration of annuity** :
- **Col116 Age of annuitants** :
- **Col117 Value of the property** :
- **Col118 Current monthly revenue** :
- **Col119 Percentage rented** :
- **Col120 Venue of the sale** :
- **Col121 Single session** :
- **Col122 Terms of visit** :
- **Col123 Starting price** :
- **Col124 EPC description** :
- **Col125 Sea view** :
- **Col126 Floor** :
- **Col127 Number of floors** :

raw_attributes_list = df.columns.tolist()
for index,attribute in enumerate(raw_attributes_list):
    values = df[attribute].unique()[:10]
    dtype = df[attribute].dtype
    nanval = df[attribute].isna().sum()
    print(f"|Col{index+1}|{attribute}|{dtype}|{nanval}|{values}| | |")

|Column|Attribute|dtype|Nan count|values|Clean_Status|Comments|
|-|-|-|-|-|-|-|
|Col1|Available as of|object|6279|['After signing the deed' 'Depending on the tenant' 'To be defined' 'Immediately' nan 'At delivery']| | |
|Col2|Available date|object|15614|['November 12 2022 - 12:00 AM' 'September 15 2022 - 12:00 AM' nan 'December 20 2022 - 12:00 AM' 'January 1 2023 - 12:00 AM'  'September 30 2023 - 12:00 AM' 'December 12 2022 - 12:00 AM'  'May 1 2023 - 12:00 AM' 'December 1 2022 - 12:00 AM'  'January 30 2023 - 12:00 AM']| | |
|Col3|Construction year|float64|6037|[1899. 1941. 1961. 1989. 1924. 1975. 1885. 1936. 1996. 1976.]| | |
|Col4|Building condition|object|3319|['Good' 'To renovate' 'As new' 'To be done up' nan 'Just renovated'  'To restore']| | |
|Col5|Street frontage width|object|13956|['5 m' '4.44 m' '7 m' '12 m' '15 m' '6.5 m' '8 m' '13 m' '6 m' '37 m']| | |
|Col6|Number of frontages|float64|4588|[ 2.  3.  4. nan  1.  6.  5.  8.]| | |
|Col7|Surroundings type|object|11641|['Isolated' 'Urban' 'Living area (residential, urban or rural)'  'Landscape' nan 'Countryside' 'Mall' 'Shop street' 'Fitted out'  'Not fitted out']| | |
|Col8|Virtual visit|object|15558|['Link to the virtual visit' nan '360° tour']| | |
|Col9|Living area|object|1374|['122 m² square meters' '119 m² square meters' '200 m² square meters'  '404 m² square meters' '350 m² square meters' '130 m² square meters'  '306 m² square meters' '1010 m² square meters' '165 m² square meters'  '125 m² square meters']| | |
|Col10|Living room surface|object|8986|['14 m² square meters' '32 m² square meters' '45 m² square meters'  '55 m² square meters' nan '35 m² square meters' '33 m² square meters'  '40 m² square meters' '22 m² square meters' '31 m² square meters']| | |
|Col11|Dining room|object|14838|['Yes' nan]| | |
|Col12|Kitchen type|object|4671|['Semi equipped' 'USA installed' 'Installed' 'Hyper equipped'  'Not installed' 'USA semi equipped' 'USA hyper equipped' nan  'USA uninstalled']| | |
|Col13|Kitchen surface|object|10529|['13 m² square meters' '10 m² square meters' '7 m² square meters'  '16 m² square meters' nan '14 m² square meters' '11 m² square meters'  '21 m² square meters' '25 m² square meters' '33 m² square meters']| | |
|Col14|Bedrooms|float64|402|[ 2.  4.  3.  6. 10.  7.  5.  8. nan  9.]| | |
|Col15|Bedroom 1 surface|object|8336|['17 m² square meters' '14 m² square meters' '20 m² square meters'  '12 m² square meters' nan '10 m² square meters' '19 m² square meters'  '36 m² square meters' '16 m² square meters' '34 m² square meters']| | |
|Col16|Bedroom 2 surface|object|9341|['13 m² square meters' '10 m² square meters' '16 m² square meters' nan  '11 m² square meters' '15 m² square meters' '17 m² square meters'  '20 m² square meters' '29 m² square meters' '19 m² square meters']| | |
|Col17|Bathrooms|float64|2430|[ 1.  2.  3.  5.  7.  4. nan  6. 11.  8.]| | |
|Col18|Shower rooms|float64|13362|[ 1. nan  2.  3.  4.  5.  6. 11.  9.  7.]| | |
|Col19|Toilets|float64|3183|[ 1.  2.  3.  5.  4.  7. nan  6. 11.  8.]| | |
|Col20|Office surface|object|16630|['32 m² square meters' nan '10 m² square meters' '13 m² square meters'  '17 m² square meters' '14 m² square meters' '18 m² square meters'  '23 m² square meters' '24 m² square meters' '16 m² square meters']| | |
|Col21|Office|object|11362|['Yes' nan 'No']| | |
|Col22|Basement|object|10983|['Yes' nan]| | |
|Col23|Attic|object|16401|['Yes' nan]| | |
|Col24|Furnished|object|4322|['No' nan 'Yes']| | |
|Col25|Surface of the plot|object|9508|['335 m² square meters' '49 m² square meters' '300 m² square meters'  '1000 m² square meters' '263 m² square meters' '991 m² square meters'  '460 m² square meters' '360 m² square meters' '16959 m² square meters'  '610 m² square meters']| | |
|Col26|Width of the lot on the street|object|15312|['5 m meters' '4.44 m meters' '10 m meters' '20 m meters' '7 m meters'  '60 m meters' '6.5 m meters' nan '275 m meters' '6 m meters']| | |
|Col27|Connection to sewer network|object|13056|['Connected' nan 'Not connected']| | |
|Col28|Gas, water &amp; electricity|object|11388|['No' 'Yes' nan]| | |
|Col29|Garden surface|object|14121|['171 m² square meters' nan '200 m² square meters' '1000 m² square meters'  '140 m² square meters' '215 m² square meters' '15324 m² square meters'  '430 m² square meters' '540 m² square meters' '1737 m² square meters']| | |
|Col30|Terrace surface|object|9872|['20 m² square meters' nan '12 m² square meters' '52 m² square meters'  '14 m² square meters' '80 m² square meters' '30 m² square meters'  '25 m² square meters' '35 m² square meters' '38 m² square meters']| | |
|Col31|Elevator|object|8368|['No' nan 'Yes']| | |
|Col32|Accessible for disabled people|object|10556|['No' nan 'Yes']| | |
|Col33|TV cable|object|12170|['Yes' nan 'No']| | |
|Col34|Swimming pool|object|10524|['No' nan 'Yes']| | |
|Col35|Primary energy consumption|object|61|['448 kWh/m² kilowatt hour per square meters'  '280 kWh/m² kilowatt hour per square meters'  '274 kWh/m² kilowatt hour per square meters' '262 kWh/m² kilowatt hour per square meters' '445 kWh/m² kilowatt hour per square meters' '482 kWh/m² kilowatt hour per square meters' '217 kWh/m² kilowatt hour per square meters' '424 kWh/m² kilowatt hour per square meters' '245 kWh/m² kilowatt hour per square meters' '649 kWh/m² kilowatt hour per square meters']| | |
|Col36|Energy class|object|61|['F' 'C' 'E' 'G' 'B' 'D' nan 'Not specified' 'A' 'A+']| | |
|Col37|Unnamed: 36|object|2796|['F' 'C' 'E' 'G' 'B' 'D' nan 'A' 'A+' 'Not specified']| | |
|Col38|Reference number of the EPC report|object|67|['20220824006893' '20220512-0002601697-RES-1' '20220815\xad-0000618777-\xad01-\xad8' '20180828-0002085595-1' 'G' '20220820003559' '2223752' '20220531-0000613819-01-6' '20220812007352' '2550943']| | |
|Col39|CO₂ emission|object|61|['111 kg CO₂/m²' '6684 kg CO₂/m²' '68 kg CO₂/m²' '92885 kg CO₂/m²' '114 kg CO₂/m²' '51 kg CO₂/m²' '1632 kg CO₂/m²' '85 kg CO₂/m²' '61 kg CO₂/m²' '627 kg CO₂/m²']| | |
|Col40|Yearly theoretical total energy consumption|object|61|['111 kWh/year' 'Not specified' '70159 kWh/year' '76890 kWh/year' '49142 kWh/year' '57454 kWh/year' '108270 kWh/year' '64571 kWh/year' '133062 kWh/year' nan]| | |
|Col41|Heating type|object|4400|['Fuel oil' 'Gas' nan 'Pellet' 'Electric' 'Carbon' 'Wood' 'Solar']| | |
|Col42|Double glazing|object|3437|['Yes' nan 'No']| | |
|Col43|Planning permission obtained|object|10813|['Yes' nan]| | |
|Col44|Total ground floor buildable|object|15671|['97 m² square meters' nan '120 m² square meters' '195 m² square meters' '348 m² square meters' '113 m² square meters' '149 m² square meters' '170 m² square meters' '200 m² square meters' '325 m² square meters']| | |
|Col45|Subdivision permit|object|11155|['No' nan 'Yes']| | |
|Col46|Possible priority purchase right|object|10017|['No' 'Yes' nan]| | |
|Col47|Proceedings for breach of planning regulations|object|11159|['No' nan 'Yes']| | |
|Col48|Flood zone type|object|8404|['Non flood zone' nan 'Recognized flood zone' 'Possible flood zone' 'Property partially or completely located in a circumscribed and recognized flood zone' 'Property partially or completely located in a possible flood zone and located in a circumscribed waterside zone' 'Circumscribed zone' 'Possible circumscribed waterside zone' 'Property partially or completely located in a circumscribed flood zone' 'Property partially or completely located in a circumscribed waterside zone']| | |
|Col49|Latest land use designation|object|9946|['Living area (residential, urban or rural)' nan 'Residential area with cultural historical aesthetic value' 'Area of landscape interest' 'Mixed living area (accommodation &amp; economic activities)' 'Natural area' 'Residential park' 'Economic activity area' 'Farming area' 'Industrial area']| | |
|Col50|Price|object|633|['Make an offer starting from € 159,000 Make an offer starting from 159000 €' '€ 345,000 345000 €' '€ 895,000 895000 €' '€ 570,000 570000 €' '€ 800,000 800000 €' '€ 425,000 425000 €' '€ 450,000 450000 €' '€ 750,000 750000 €' 'Make an offer starting from € 849,000 Make an offer starting from 849000 €' '€ 129,000 129000 €']| | |
|Col51|Cadastral income|object|8254|['€ 441 441 €' '€ 417 417 €' '€ 2,627 2627 €' '€ 1,700 1700 €' '€ 2,662 2662 €' '€ 780 780 €' '€ 1,690 1690 €' '€ 2,625 2625 €' '€ 1,456 1456 €' '€ 349 349 €']| | |
|Col52|Tenement building|object|18385|[nan 'No' 'Yes']| | |
|Col53|Address|object|1024|['Rue de la Wallonie 2A 4680 - Oupeye' 'Sint-Denijslaan 1 9000 - Gent' 'Wezembeeklaan, 120 1950 - Kraainem' nan 'Rue Emile Bouilliot 61 1050 - Ixelles' 'Chaussée de Liège 7 6900 - Marche' 'Stationstraat 30 9600 - Ronse' "Avenue Baron Albert d'Huart 130 1950 - Kraainem" 'Chaussée de Tirlemont 15/1 4260 - Braives' 'Avenue Fond Jean Rosy 17A 1330 - Rixensart']| | |
|Col54|Website|object|4154|['http://www.nigel-immo.be' 'http://www.immodavinci.be' 'http://www.keystones.be' nan 'http://www.caron-realestate.be' 'https://honesty.be/' 'http://www.latouretpetit.be' 'http://www.immodemarneffe.be' 'http://www.weinvest.be' 'http://www.expertissimmo.eu']| | |
|Col55|External reference|object|3480|['4862172' '4964255' '4964425' nan '4964294' '4961946' '4862197' '4962685' '4861727' '4961494']| | |
|Col56|subtype_of_property|object|0|['house' 'villa' 'apartment-block' 'exceptional-property' 'country-cottage' 'mixed-use-building' 'other-property' 'manor-house' 'farmhouse' 'town-house']| | |
|Col57|type_of_property|object|0|['house' 'other' 'apartment']| | |
|Col58|type_of_sale|object|0|['for-sale']| | |
|Col59|locality|object|0|['ans' 'gent' 'woluwe-saint-pierre' 'galmaarden' 'forest' 'waimes' 'ronse' 'dalhem' 'remicourt' 'chaumont-gistoux']| | |
|Col60|zip_code|int64|0|[4430 9000 1150 1570 1190 4950 9600 4607 4350 1325]| | |
|Col61|url|object|0|['https://www.immoweb.be/en/classified/house/for-sale/ans/4430/10147032?searchId=633aeaf1e42e4' 'https://www.immoweb.be/en/classified/house/for-sale/gent/9000/10146903?searchId=633aeaf1e42e4' 'https://www.immoweb.be/en/classified/villa/for-sale/woluwe-saint-pierre/1150/10146862?searchId=633aeaf1e42e4' 'https://www.immoweb.be/en/classified/villa/for-sale/galmaarden/1570/9935270?searchId=633aeaf1e42e4' 'https://www.immoweb.be/en/classified/apartment-block/for-sale/forest/1190/10145907?searchId=633aeaf1e42e4' 'https://www.immoweb.be/en/classified/house/for-sale/waimes/4950/10143254?searchId=633aeaf1e42e4' 'https://www.immoweb.be/en/classified/apartment-block/for-sale/ronse/9600/10143136?searchId=633aeaf1e42e4' 'https://www.immoweb.be/en/classified/house/for-sale/woluwe-saint-pierre/1150/10142887?searchId=633aeaf1e42e4' 'https://www.immoweb.be/en/classified/exceptional-property/for-sale/dalhem/4607/10143366?searchId=633aeaf1e42e4' 'https://www.immoweb.be/en/classified/house/for-sale/ronse/9600/10140837?searchId=633aeaf1e42e4']| | |
|Col62|Bedroom 3 surface|object|12560|[nan '14 m² square meters' '10 m² square meters' '27 m² square meters' '12 m² square meters' '8 m² square meters' '13 m² square meters' '15 m² square meters' '17 m² square meters' '21 m² square meters']| | |
|Col63|Bedroom 4 surface|object|15725|[nan '15 m² square meters' '8 m² square meters' '16 m² square meters' '14 m² square meters' '11 m² square meters' '18 m² square meters' '23 m² square meters' '10 m² square meters' '12 m² square meters']| | |
|Col64|Neighbourhood or locality|object|15635|[nan 'Omgeving Kraainem' 'Brussels' 'Zone Province de Liège' 'Basse Meuse' 'Waremme et 5 kms' 'Brabant Wallon' 'Bruxelles Sud' 'Les Woluwes' 'Bruxelles Est']| | |
|Col65|Covered parking spaces|float64|10876|[nan  1.  2. 10.  5.  4.  3.  7. 12.  6.]| | |
|Col66|Outdoor parking spaces|float64|12454|[nan  1.  2.  3.  6.  8.  5.  4. 20.  7.]| | |
|Col67|Dressing room|object|14166|[nan 'No' 'Yes']| | |
|Col68|Laundry room|object|16928|[nan 'Yes']| | |
|Col69|Professional space|object|17380|[nan 'Yes' 'No']| | |
|Col70|Basement surface|object|16570|[nan '148 m² square meters' '39 m² square meters' '49 m² square meters' '60 m² square meters' '48 m² square meters' '2 m² square meters' '23 m² square meters' '15 m² square meters' '50 m² square meters']| | |
|Col71|Attic surface|object|17738|[nan '36 m² square meters' '66 m² square meters' '35 m² square meters' '124 m² square meters' '61 m² square meters' '4 m² square meters' '60 m² square meters' '52 m² square meters' '47 m² square meters']| | |
|Col72|Armored door|object|14419|[nan 'No' 'Yes']| | |
|Col73|Land is facing street|object|17034|[nan 'No' 'Yes']| | |
|Col74|Wooded land|object|17110|[nan 'No' 'Yes']| | |
|Col75|Plot at rear|object|17084|[nan 'No' 'Yes']| | |
|Col76|Flat land|object|16988|[nan 'Yes' 'No']| | |
|Col77|Caretaker|object|15128|[nan 'No' 'Yes']| | |
|Col78|Intercom|object|12766|[nan 'No' 'Yes']| | |
|Col79|Secure access / alarm|object|13389|[nan 'Yes' 'No']| | |
|Col80|Air conditioning|object|17603|[nan 'No' 'Yes']| | |
|Col81|Visio phone|object|12905|[nan 'No' 'Yes']| | |
|Col82|Jacuzzi|object|17314|[nan 'No' 'Yes']| | |
|Col83|Sauna|object|17311|[nan 'No' 'Yes']| | |
|Col84|Internet|object|17620|[nan 'Yes' 'No']| | |
|Col85|Conformity certification for fuel tanks|object|17413|[nan 'Yes, conform' 'No' 'Yes, not conform']| | |
|Col86|Heat pump|object|15254|[nan 'No' 'Yes']| | |
|Col87|Photovoltaic solar panels|object|16719|[nan 'Yes' 'No']| | |
|Col88|Thermic solar panels|object|16039|[nan 'Yes' 'No']| | |
|Col89|Common water heater|object|16844|[nan 'No' 'Yes']| | |
|Col90|Type of building|object|17393|[nan 'All kind' 'Villa' 'Bungalow' 'Apartment building' 'Town-house']| | |
|Col91|Terrace|object|14309|[nan 'Yes']| | |
|Col92|Bedroom 5 surface|object|17323|[nan '10 m² square meters' '13 m² square meters' '16 m² square meters' '15 m² square meters' '14 m² square meters' '23 m² square meters' '17 m² square meters' '11 m² square meters' '18 m² square meters']| | |
|Col93|Professional space surface|object|18039|[nan '167 m² square meters' '40 m² square meters' '340 m² square meters' '73 m² square meters' '30 m² square meters' '60 m² square meters' '19 m² square meters' '83 m² square meters' '36 m² square meters']| | |
|Col94|As built plan|object|12584|[nan 'Yes, conform' 'No']| | |
|Col95|Living room|object|15544|[nan 'Yes']| | |
|Col96|How many fireplaces?|float64|17484|[nan  1.  2.  3. 10.  6.  4.  5. 40.]| | |
|Col97|Garden|object|16266|[nan 'Yes']| | |
|Col98|Garden orientation|object|14815|[nan 'South East' 'South' 'North' 'West' 'North East' 'South West' 'North West' 'East']| | |
|Col99|Terrace orientation|object|17243|[nan 'South' 'South West' 'North East' 'West' 'South East' 'East' 'North' 'North West']| | |
|Col100|E-level (overall energy performance)|float64|16920|[   nan 54901.   332. 88785.   181.   131.   396. 17017. 62189. 76408.]| | |
|Col101|Agent's name|object|17451|[nan 'Cédric Mathonnet' 'Immo TK' 'Alicia MOREELS' 'Alexis Soillle' 'Vranckx Gauthier' 'Clara MALAXOS' 'Desira Kevin' 'NINETEEN PROPERTY' 'HICKL-SZABO, Sonja']| | |
|Col102|Number of annexes|float64|18333|[nan  1.  2.  6.  5.  3.]| | |
|Col103|Isolated|object|18291|[nan 'Yes']| | |
|Col104|Obligation to build|object|16149|[nan 'No' 'Yes']| | |
|Col105|Monthly charges|object|17488|[nan '€ 60 60 €' '€ 100 100 €' '€ 50 50 €' '€ 350 350 €' '€ 21 21 €' '€ 75 75 €' '€ 290 290 €' '€ 107 107 €' '€ 30 30 €']| | |
|Col106|Property name|object|17361|[nan "PORTE D'ANDERLECHT - ABATTOIRS" '-' 'PARC TENBOSCH - CHATELAIN - LEPOUTRE' 'MONTGOMERY - PARC DE TERVUREN - MERODE' 'RUBENSLAAN 5' 'Uiterste verzorgde villa met magnifieke tuin!' 'Patrick Van Humbeek' 'SERGEANT DE BRUYNESTRAAT 32A' 'Banlieu wijk']| | |
|Col107|Extra information|object|18463|[nan 'Ok' 'Each house has a private garden. A common plot allows access to vehicles behind the houses.' 'Demander les RU' 'Pleasant little garden situated next to a quiet stream.' 'Yes' 'vg, wg, gdv, gvkr, vv' 'Vg, Wg, Gvkr, Gmo, Gvv' 'Information from the Town Planning Services is available.' 'Good']| | |
|Col108|Lump sum|object|18432|[nan '€ 30,000 30000€' '€ 35,000 35000€' '€ 25,000 25000€' '€ 50,000 50000€' '€ 60,000 60000€' '€ 149,000 149000€' '€ 40,000 40000€' '€ 259,000 259000€' '€ 150,000 150000€']| | |
|Col109|Monthly annuity|object|18432|[nan '€ 580 580€' '€ 550 550€' '€ 660 660€' '€ 950 950€' '€ 800 800€' '€ 1,170 1170€' '€ 1 1€' '€ 535 535€' '€ 1,665 1665€']| | |
|Col110|Indexed annuity|object|18432|[nan 'Yes' 'No']| | |
|Col111|Reversionary annuity|object|18433|[nan 'No' 'Yes']| | |
|Col112|Bare ownership sale|object|18432|[nan 'No' 'Yes']| | |
|Col113|Number of annuitants|float64|18440|[nan  1.  2.]| | |
|Col114|Age of annuitant|float64|18474|[nan 75. 65. 76. 70.]| | |
|Col115|Maximum duration of annuity|float64|18479|[nan]| | |
|Col116|Age of annuitants|object|18474|[nan '76, 73' '72, 75' '71, 76' '75, 74' '85, 82']| | |
|Col117|Value of the property|object|18443|[nan '€ 280,000 280000€' '€ 60,000 60000€' '€ 245,000 245000€' '€ 110,000 110000€' '€ 220,000 220000€' '€ 395,000 395000€' '€ 420,000 420000€' '€ 40,000 40000€' '€ 260,000 260000€']| | |
|Col118|Current monthly revenue|float64|18479|[nan]| | |
|Col119|Percentage rented|float64|18470|[ nan 100.  20.   1.]| | |
|Col120|Venue of the sale|object|18047|[nan 'Link to online sale']| | |
|Col121|Single session|object|18047|[nan 'No']| | |
|Col122|Terms of visit|object|18072|[nan 'fr : SANS RENDEZ-VOUS : Je 13/10 de 17h à 19h, sa 15/10 de 14h à 16h, je 20/10 de 17h à 19h, sa 22/10 de 14h à 16h, je 27/10 de 17h à 19h, sa 05/11 de 14h à 16h, je 10/11 de 17h à 19h, sa 12/11 de 14h à 16h, lu 14/11 de 10h à 12h, je 17/11 de 09h à 11h.' 'fr : Visite sur prise de rendez-vous avec Madame VAN DEN HOVE : 082/77.18.77 - Tapez 4' 'nl : Bezoek na afspraak op: - Woensdag 28 september, 5, 12 en 19 oktober 2022 van 16u00 tot 18u00; - Zaterdag 1, 8, 15 en 22 oktober 2022 van 10u00 tot 12u00.' 'fr : Demande de visite sous le numéro 087 / 89 18 16' 'nl : OP AFSPRAAK, op zondag 18 september en zaterdag 1 oktober. Andere bezoeken kunnen steeds op verzoek ingepland worden ([email\xa0protected] of 09/388.82.77)' "fr : Sur rendez-vous via l'étude (061/21.11.22)" 'nl : Het goed zal kunnen bezocht worden van 14.00 tot 16.00 uur op: woe. 21 september 2022; za. 24 september 2022; woe. 28 september 2022; za. 1 oktober 2022; woe. 5 oktober 2022; za. 8 oktober 2022; woe. 12 oktober 2022; za. 15 oktober 2022' 'nl : Het goed zal door de kandidaat-kopers kunnen bezocht worden op afspraak; op zondag 4 en 18 september' 'fr : sam. 24/09, 01-08-15/10 et jeu. 29/09, 06-13/10 de 11-13h. nl : zat. 24/09, 01-08-15/10 en don. 29/09, 06-13/10 van 11-13u.']| | |
|Col123|Starting price|object|18047|[nan '€ 150,000 150000 €' '€ 100,000 100000 €' '€ 395,000 395000 €' '€ 170,000 170000 €' '€ 209,000 209000 €' '€ 120,000 120000 €' '€ 165,000 165000 €' '€ 276,000 276000 €' '€ 382,750 382750 €']| | |
|Col124|EPC description|object|18474|[nan "the EPC result is E as we couldn't find the invoices of the new roof from 2004 with isolation so it should be better than E as mentioned on the certificate" 'Very nice flat 3 rooms, 1 garage, spacious torage room, marble floor and stairs beautiful and large terrace of 12 m2' 'Gasjaarverbruik in 2021: 150 euro / jaarElektriciteitsverbruik in 2021: 45e / maand' 'Ongoing' 'Le PEB “classé A“ n° 20220727020160 indique une consommation d’énergie primaire de 80 KWH /m² par an et d’énergie primaire théorique total de 7402 KWH/an.Le PEB indique des besoins en chauffage MINIMES, une EXCELLENTE installation de chauffage et d’eau chaude sanitaire et un système de ventilation COMPLET. Le chauffage et l’eau chaude sanitaire instantanée sont assurés par une chaudière gaz à condensation de la marque Viessmann Vitodens 200 W de 26 KW avec option régulation par sonde extérieur installée en 2010. (Installation et entretient périodique réalisé par la société Miessen à Embourg).']| | |
|Col125|Sea view|object|18454|[nan 'Yes']| | |
|Col126|Floor|float64|12078|[ nan   1.   2.   5.   3. 999.   7.   4.  19.   6.]| | |
|Col127|Number of floors|float64|12678|[nan  2.  5.  4.  1.  3.  6. 12.  7. 26.]| | |

print(df['Building condition'].unique())

**dtype cleaning**

In [4]:
def remove_str(strig_to_remove):
    df['Street frontage width'] = df['Street frontage width'].str.replace(strig_to_remove,'').fillna(0).astype(float)

In [5]:
df['Street frontage width'] = df['Street frontage width'].str.replace(' m','')  #removes text
df['Street frontage width'] = df['Street frontage width'].fillna(0)  #fills missing data with 0
df['Street frontage width'] = df['Street frontage width'].astype(float)  #cast float type

In [6]:
df['Street frontage width']

0         5.00
1         4.44
2         7.00
3        12.00
4         7.00
         ...  
18474     0.00
18475     0.00
18476     0.00
18477     0.00
18478     0.00
Name: Street frontage width, Length: 18479, dtype: float64

In [28]:
df['Swimming pool'].fillna(0).unique()
#df['Swimming pool'].unique()

array(['No', 0, 'Yes'], dtype=object)

In [29]:
df['Swimming pool'] = df['Swimming pool'].map({'Yes': 1, 'No': 0}).fillna(0)
df['Swimming pool'].unique()

array([0., 1.])

In [37]:
df['Kitchen type'].fillna(0).unique()

array([0.5 , 0.75, 1.  , 0.  ])

In [52]:
df['Kitchen type'] = df['Kitchen type'].map({'Semi equipped': 0.5, 'USA installed': 0.75, 'Installed': 0.75, 'Hyper equipped': 1.0, 'Not installed': 0.0, 'USA semi equipped': 0.5, 'USA hyper equipped': 1.0,  'USA uninstalled': 0.75})
df['Kitchen type'].fillna(0).unique().astype(float)

array([0.5 , 0.75, 1.  , 0.  ])

In [50]:
df['Kitchen type'].dtype

dtype('float64')

## 3. Data exploration

## Conclusions